### ISSUE TITLE
Meteor code must always run within a Fiber error. Mongo crashed after ENOENT

### ISSUE THREAD

**Issue by richardwlu** [RocketChat/Rocket.Chat#1858]
Mongo recently crashed after an unexpected error which prevented users from logging into the Rocket.chat client, though the UI was accessible. Mongo showed it restarted itself but was in an unstable state. A systemctl restart of mongo allowed for it to continue running. I am looking deeper to see if I can find the root cause and would love some insight.

The mongo log seems to show there was an error with writing to a log file:

	2016-01-10T09:00:00.268+0000 I QUERY    [conn81] query rocketchat.rocketchat_message query: { $query: { ts: { $exists: 1 } }, orderby: { ts: -1 } } planSummary: COLLSCAN, COLLSCAN ntoreturn:1 ntoskip:0 nscanned:0 nscannedObjects:50752 scanAndOrder:1 keyUpdates:0 writeConflicts:0 numYields:396 nreturned:1 reslen:163 locks:{ Global: { acquireCount: { r: 794 } }, MMAPV1Journal: { acquireCount: { r: 397 } }, Database: { acquireCount: { r: 397 } }, Collection: { acquireCount: { R: 397 } } } 168ms
	2016-01-10T09:08:16.861+0000 I CONTROL  [journal writer] LogFile::synchronousAppend failed with 8192 bytes unwritten out of 8192 bytes;  b=0x5052000 errno:5 Input/output error
	2016-01-10T09:08:16.861+0000 I -        [journal writer] Fatal Assertion 13515
	2016-01-10T09:08:16.866+0000 I CONTROL  [journal writer]
	 0xf9cfb2 0xf3a239 0xf1e3e6 0xf3acce 0xd56953 0xd56ba5 0xd58c7c 0xfeb52b 0x7f8b62152dc5 0x7f8b60be121d
	----- BEGIN BACKTRACE -----
	{"backtrace":[{"b":"400000","o":"B9CFB2"},{"b":"400000","o":"B3A239"},{"b":"400000","o":"B1E3E6"},{"b":"400000","o":"B3ACCE"},{"b":"400000","o":"956953"},{"b":"400000","o":"956BA5"},{"b":"400000","o":"958C7C"},{"b":"400000","o":"BEB52B"},{"b":"7F8B6214B000","o":"7DC5"},{"b":"7F8B60AEB000","o":"F621D"}],"processInfo":{ "mongodbVersion" : "3.0.8", "gitVersion" : "83d8cc25e00e42856924d84e220fbe4a839e605d", "uname" : { "sysname" : "Linux", "release" : "3.10.0-327.el7.x86_64", "version" : "#1 SMP Thu Oct 29 17:29:29 EDT 2015", "machine" : "x86_64" }, "somap" : [ { "elfType" : 2, "b" : "400000", "buildId" : "073CFAD0CAAF2E7E6F295A10384EC3864A39B0F9" }, { "b" : "7FFFED98B000", "elfType" : 3, "buildId" : "17A121B1F7BBB010F54735FFDE3347B27B33884D" }, { "b" : "7F8B6214B000", "path" : "/lib64/libpthread.so.0", "elfType" : 3, "buildId" : "0D3AFB2828490FDE7B2BE3BEEBACBDB4D94A999F" }, { "b" : "7F8B61EDB000", "path" : "/lib64/libssl.so.10", "elfType" : 3, "buildId" : "8DB4545998776514159031B754BB67F7F396F83A" }, { "b" : "7F8B61AF3000", "path" : "/lib64/libcrypto.so.10", "elfType" : 3, "buildId" : "038F79F7C3F6E60C29184B8E70D0B1E62525D64D" }, { "b" : "7F8B618EB000", "path" : "/lib64/librt.so.1", "elfType" : 3, "buildId" : "CCC119FE8F4D8D262AFC67DDC36C2F266F30586F" }, { "b" : "7F8B616E3000", "path" : "/lib64/libdl.so.2", "elfType" : 3, "buildId" : "DECA756A631284EEE020E3AE1CADCE9BDFB9914D" }, { "b" : "7F8B613D3000", "path" : "/lib64/libstdc++.so.6", "elfType" : 3, "buildId" : "74AAD82C77F6D211BD9E4DA27BF4F62A0D938C4B" }, { "b" : "7F8B610CB000", "path" : "/lib64/libm.so.6", "elfType" : 3, "buildId" : "F129A8EBE4CD7D042019BFACD9A58677C18F2AE3" }, { "b" : "7F8B60EB3000", "path" : "/lib64/libgcc_s.so.1", "elfType" : 3, "buildId" : "97D5E2F5739B715C3A0EC9F95F7336E232346CA8" }, { "b" : "7F8B60AEB000", "path" : "/lib64/libc.so.6", "elfType" : 3, "buildId" : "508F7FF5F9802FE2FE52FD9F5D19D0D732D55A56" }, { "b" : "7F8B6236B000", "path" : "/lib64/ld-linux-x86-64.so.2", "elfType" : 3, "buildId" : "09E1BB4D034C7263810A41100647068858A7ECB6" }, { "b" : "7F8B6089B000", "path" : "/lib64/libgssapi_krb5.so.2", "elfType" : 3, "buildId" : "8AB5682155DE13D0916B984306B4E044E216B2EB" }, { "b" : "7F8B605B3000", "path" : "/lib64/libkrb5.so.3", "elfType" : 3, "buildId" : "BE8968836D439581B2816CE3827642FCF4B8BF4A" }, { "b" : "7F8B603AB000", "path" : "/lib64/libcom_err.so.2", "elfType" : 3, "buildId" : "B25574847B066A26CD593C8101DF6779898FF2C2" }, { "b" : "7F8B60173000", "path" : "/lib64/libk5crypto.so.3", "elfType" : 3, "buildId" : "F5784ED7E64118BAFE898DBF178DC9E37CBDA4AA" }, { "b" : "7F8B5FF5B000", "path" : "/lib64/libz.so.1", "elfType" : 3, "buildId" : "FC37913FB197B822BCDBF3697D061E248698CEC1" }, { "b" : "7F8B5FD4B000", "path" : "/lib64/libkrb5support.so.0", "elfType" : 3, "buildId" : "4BBED12CFDC9647C8771A4B897E0D5A4F217ED7C" }, { "b" : "7F8B5FB43000", "path" : "/lib64/libkeyutils.so.1", "elfType" : 3, "buildId" : "8CA73C16CFEB9A8B5660015B9223B09F87041CAD" }, { "b" : "7F8B5F923000", "path" : "/lib64/libresolv.so.2", "elfType" : 3, "buildId" : "262D22F4D416B270C2AA70EF5FB2100E12796A3F" }, { "b" : "7F8B5F6FB000", "path" : "/lib64/libselinux.so.1", "elfType" : 3, "buildId" : "5062031216B995004A297D555D834C0109F7598C" }, { "b" : "7F8B5F493000", "path" : "/lib64/libpcre.so.1", "elfType" : 3, "buildId" : "8E3819A80BF876382A6F0CB2A08F82F1742EE8DB" }, { "b" : "7F8B5F26B000", "path" : "/lib64/liblzma.so.5", "elfType" : 3, "buildId" : "61D7D46225E85F144221E1424B87FBF3CB2B9D3F" } ] }}
	 mongod(_ZN5mongo15printStackTraceERSo+0x32) [0xf9cfb2]
	 mongod(_ZN5mongo10logContextEPKc+0xE9) [0xf3a239]
	 mongod(_ZN5mongo13fassertFailedEi+0x66) [0xf1e3e6]
	 mongod(_ZN5mongo7LogFile17synchronousAppendEPKvm+0x21E) [0xf3acce]
	 mongod(_ZN5mongo3dur7Journal7journalERKNS0_11JSectHeaderERKNS_14AlignedBuilderE+0x203) [0xd56953]
	 mongod(_ZN5mongo3dur14WRITETOJOURNALERKNS0_11JSectHeaderERKNS_14AlignedBuilderE+0x35) [0xd56ba5]
	 mongod(_ZN5mongo3dur13JournalWriter20_journalWriterThreadEv+0x33C) [0xd58c7c]
	 mongod(+0xBEB52B) [0xfeb52b]
	 libpthread.so.0(+0x7DC5) [0x7f8b62152dc5]
	 libc.so.6(clone+0x6D) [0x7f8b60be121d]
	-----  END BACKTRACE  -----
	2016-01-10T09:08:16.866+0000 I -        [journal writer]

	***aborting after fassert() failure

	2016-01-10T14:09:23.786+0000 I CONTROL  ***** SERVER RESTARTED *****
	2016-01-10T14:09:23.793+0000 I CONTROL  [initandlisten] MongoDB starting : pid=23308 port=27017 dbpath=/var/lib/mongo 64-bit host=chat.zayo.us
	2016-01-10T14:09:23.793+0000 I CONTROL  [initandlisten] db version v3.0.8
	2016-01-10T14:09:23.793+0000 I CONTROL  [initandlisten] git version: 83d8cc25e00e42856924d84e220fbe4a839e605d
	2016-01-10T14:09:23.793+0000 I CONTROL  [initandlisten] build info: Linux ip-10-154-248-24 3.4.43-43.43.amzn1.x86_64 #1 SMP Mon May 6 18:04:41 UTC 2013 x86_64 BOOST_LIB_VERSION=1_49
	2016-01-10T14:09:23.793+0000 I CONTROL  [initandlisten] allocator: tcmalloc
	2016-01-10T14:09:23.793+0000 I CONTROL  [initandlisten] options: { config: "/etc/mongod.conf", net: { bindIp: "127.0.0.1", port: 27017 }, processManagement: { fork: true, pidFilePath: "/var/run/mongodb/mongod.pid" }, replication: { replSetName: "001-rs" }, storage: { dbPath: "/var/lib/mongo", journal: { enabled: true } }, systemLog: { destination: "file", logAppend: true, path: "/var/log/mongodb/mongod.log" } }
	2016-01-10T14:09:23.825+0000 W -        [initandlisten] Detected unclean shutdown - /var/lib/mongo/mongod.lock is not empty.
	2016-01-10T14:09:23.834+0000 I STORAGE  [initandlisten]
	2016-01-10T14:09:23.834+0000 I STORAGE  [initandlisten] ** WARNING: Readahead for /var/lib/mongo is set to 4096KB
	2016-01-10T14:09:23.834+0000 I STORAGE  [initandlisten] **          We suggest setting it to 256KB (512 sectors) or less
	2016-01-10T14:09:23.834+0000 I STORAGE  [initandlisten] **          http://dochub.mongodb.org/core/readahead
	2016-01-10T14:09:23.834+0000 I JOURNAL  [initandlisten] journal dir=/var/lib/mongo/journal
	2016-01-10T14:09:23.834+0000 I JOURNAL  [initandlisten] recover begin
	2016-01-10T14:09:23.834+0000 I JOURNAL  [initandlisten] recover lsn: 945751452
	2016-01-10T14:09:23.834+0000 I JOURNAL  [initandlisten] recover /var/lib/mongo/journal/j._4
	2016-01-10T14:09:23.836+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731146075 < lsn:945751452
	2016-01-10T14:09:23.852+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731205355 < lsn:945751452
	2016-01-10T14:09:23.855+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731264795 < lsn:945751452
	2016-01-10T14:09:23.855+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731324165 < lsn:945751452
	2016-01-10T14:09:23.855+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731383565 < lsn:945751452
	2016-01-10T14:09:23.856+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731442985 < lsn:945751452
	2016-01-10T14:09:23.856+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731502305 < lsn:945751452
	2016-01-10T14:09:23.862+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731561415 < lsn:945751452
	2016-01-10T14:09:23.869+0000 I JOURNAL  [initandlisten] recover skipping application of section seq:731620595 < lsn:945751452
	2016-01-10T14:09:23.872+0000 I JOURNAL  [initandlisten] recover skipping application of section more...
	2016-01-10T14:09:29.214+0000 I JOURNAL  [initandlisten] recover /var/lib/mongo/journal/j._5
	2016-01-10T14:09:31.137+0000 I JOURNAL  [initandlisten] recover cleaning up
	2016-01-10T14:09:31.137+0000 I JOURNAL  [initandlisten] removeJournalFiles
	2016-01-10T14:09:31.314+0000 I JOURNAL  [initandlisten] recover done
	2016-01-10T14:09:31.778+0000 I JOURNAL  [durability] Durability thread started
	2016-01-10T14:09:31.778+0000 I JOURNAL  [journal writer] Journal writer thread started
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten]
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten] ** WARNING: /sys/kernel/mm/transparent_hugepage/enabled is 'always'.
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten] **        We suggest setting it to 'never'
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten]
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten] ** WARNING: /sys/kernel/mm/transparent_hugepage/defrag is 'always'.
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten] **        We suggest setting it to 'never'
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten]
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten] ** WARNING: soft rlimits too low. rlimits set to 4096 processes, 64000 files. Number of processes should be at least 32000 : 0.5 times number of files.
	2016-01-10T14:09:31.782+0000 I CONTROL  [initandlisten]

and the rocket.chat log shows an ENOENT and throws the Meteor code must always run within a Fiber error:

    [richlu@chat rocket.chat]$ cat err-0.log
    Error: ENOENT, open '/tmp/ufs/HHTCTRvN4up7S7tSH'

    /var/www/rocket.chat/bundle/programs/server/packages/meteor.js:974
        throw new Error("Meteor code must always run within a Fiber. " +
              ^
    Error: Meteor code must always run within a Fiber. Try wrapping callbacks that you pass to non-Meteor libraries with Meteor.bindEnvironment.
        at Object.Meteor._nodeCodeMustBeInFiber (packages/meteor/dynamics_nodejs.js:9:1)
        at [object Object]._.extend.get (packages/meteor/dynamics_nodejs.js:21:1)
        at Object.Meteor.isRestricted (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:137:1)
        at [object Object].Mongo.Collection.(anonymous function) [as update] (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:268:1)
        at Object.UserPresence.removeConnectionsByInstanceId (packages/konecty_user-presence/packages/konecty_user-presence.js:88:1)
        at process.<anonymous> (packages/konecty_user-presence/packages/konecty_user-presence.js:223:1)
        at process.emit (events.js:117:20)
        at process.exit (node.js:740:17)
        at process.catchException (/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/notify.js:52:15)
        at process.g (events.js:180:16)
    Error: ENOENT, open '/tmp/ufs/HHTCTRvN4up7S7tSH'

    /var/www/rocket.chat/bundle/programs/server/packages/meteor.js:974
        throw new Error("Meteor code must always run within a Fiber. " +
              ^
    Error: Meteor code must always run within a Fiber. Try wrapping callbacks that you pass to non-Meteor libraries with Meteor.bindEnvironment.
        at Object.Meteor._nodeCodeMustBeInFiber (packages/meteor/dynamics_nodejs.js:9:1)
        at [object Object]._.extend.get (packages/meteor/dynamics_nodejs.js:21:1)
        at Object.Meteor.isRestricted (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:137:1)
        at [object Object].Mongo.Collection.(anonymous function) [as update] (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:268:1)
        at Object.UserPresence.removeConnectionsByInstanceId (packages/konecty_user-presence/packages/konecty_user-presence.js:88:1)
        at process.<anonymous> (packages/konecty_user-presence/packages/konecty_user-presence.js:223:1)
        at process.emit (events.js:117:20)
        at process.exit (node.js:740:17)
        at process.catchException (/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/notify.js:52:15)
        at process.g (events.js:180:16)
    Error: No replica set primary available for query with ReadPreference PRIMARY
        at Object.Future.wait (/var/www/rocket.chat/bundle/programs/server/node_modules/fibers/future.js:395:16)
        at _Class.findOneSync (packages/meteor/helpers.js:119:1)
        at _Class.RocketChatFile.GridFS._Class.findOne (packages/rocketchat_file/file.server.coffee:49:15)
        at _Class.RocketChatFile.GridFS._Class.getFileWithReadStream (packages/rocketchat_file/file.server.coffee:87:15)
        at Object.Package [as handle] (packages/rocketchat_assets/server/assets.coffee:137:34)
        at next (/var/www/rocket.chat/bundle/programs/server/npm/webapp/node_modules/connect/lib/proto.js:190:15)
        at middleware (packages/steffo_meteor-accounts-saml/saml_server.js:174:1)
        at packages/steffo_meteor-accounts-saml/saml_server.js:164:1
        - - - - -
        at [object Object].ReplSet.checkoutReader (/var/www/rocket.chat/bundle/programs/server/npm/npm-mongo/node_modules/mongodb/lib/mongodb/connection/repl_set/repl_set.js:619:14)
        at Cursor.nextObject (/var/www/rocket.chat/bundle/programs/server/npm/npm-mongo/node_modules/mongodb/lib/mongodb/cursor.js:748:48)
        at Collection.findOne (/var/www/rocket.chat/bundle/programs/server/npm/npm-mongo/node_modules/mongodb/lib/mongodb/collection/query.js:162:10)
        at _Class.findOneSync (packages/meteor/helpers.js:118:1)
        at _Class.RocketChatFile.GridFS._Class.findOne (packages/rocketchat_file/file.server.coffee:49:15)
        at _Class.RocketChatFile.GridFS._Class.getFileWithReadStream (packages/rocketchat_file/file.server.coffee:87:15)
        at Object.Package [as handle] (packages/rocketchat_assets/server/assets.coffee:137:34)
        at next (/var/www/rocket.chat/bundle/programs/server/npm/webapp/node_modules/connect/lib/proto.js:190:15)
        at middleware (packages/steffo_meteor-accounts-saml/saml_server.js:174:1)
        at packages/steffo_meteor-accounts-saml/saml_server.js:164:1

Does anyone have an idea of why this would happen and to help prevent it?

**Comment by geekgonecrazy** [comment]
@richardwlu looks like this is having issues with the /tmp folder on the file system. Are you using GRIDFS or the filesystem for uploads?

**Comment by richardwlu** [comment]
We are currently using GRIDFS with default settings.

**Comment by richardwlu** [comment]
@geekgonecrazy We just switched to FileSystem and I uploaded an avatar and then it crashed rocketchat and it is currently having major issues with connecting and dropping every few seconds.

	PM2: 2016-01-12 19:18:49: App name:rocket.chat id:0 online

	rocket.chat-0 (err):     at process.catchException (/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/notify.js:52:15)
	rocket.chat-0 (err):     at process.g (events.js:180:16)
	rocket.chat-0 (err): Error: spawn Unknown system errno 7
	rocket.chat-0 (err):     at errnoException (child_process.js:1011:11)
	rocket.chat-0 (err):     at Process.ChildProcess._handle.onexit (child_process.js:802:34)
	rocket.chat-0 (err):
	rocket.chat-0 (err): /var/www/rocket.chat/bundle/programs/server/packages/meteor.js:974
	rocket.chat-0 (err):     throw new Error("Meteor code must always run within a Fiber. " +
	rocket.chat-0 (err):           ^
	rocket.chat-0 (err): Error: Meteor code must always run within a Fiber. Try wrapping callbacks that you pass to non-Meteor libraries with Meteor.bindEnvironment.
	rocket.chat-0 (err):     at Object.Meteor._nodeCodeMustBeInFiber (packages/meteor/dynamics_nodejs.js:9:1)
	rocket.chat-0 (err):     at [object Object]._.extend.get (packages/meteor/dynamics_nodejs.js:21:1)
	rocket.chat-0 (err):     at Object.Meteor.isRestricted (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:137:1)
	rocket.chat-0 (err):     at [object Object].Mongo.Collection.(anonymous function) [as update] (packages/dispatch_run-as-user/packages/dispatch_run-as-user.js:268:1)
	rocket.chat-0 (err):     at Object.UserPresence.removeConnectionsByInstanceId (packages/konecty_user-presence/packages/konecty_user-presence.js:88:1)
	rocket.chat-0 (err):     at process.<anonymous> (packages/konecty_user-presence/packages/konecty_user-presence.js:223:1)
	rocket.chat-0 (err):     at process.emit (events.js:117:20)
	rocket.chat-0 (err):     at process.exit (node.js:740:17)
	rocket.chat-0 (err):     at process.catchException (/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/notify.js:52:15)
	rocket.chat-0 (err):     at process.g (events.js:180:16)

	rocket.chat-0 (out): Updating login service Accounts_OAuth_Google
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Google_id
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Google_secret
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Linkedin
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Linkedin_id
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Linkedin_secret
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Meteor
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Meteor_id
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Meteor_secret
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Twitter
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Twitter_id
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Twitter_secret
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Wordpress
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Wordpress_id
	rocket.chat-0 (out): Updating login service Accounts_OAuth_Wordpress_secret
	rocket.chat-0 (out): ufs: created tmpDir /tmp/ufs
	rocket.chat-0 (out): Kadira: completed instrumenting the app
	rocket.chat-0 (out): configuring push
	rocket.chat-0 (out): Using GridFS for Avatar storage
	rocket.chat-0 (out): {"line":"80","file":"migrations_server.js","message":"Migrations: Not migrating, already at version 25","time":{"$date":1452626334892},"level":"info"}

	[PM2] Streaming realtime logs for [all] processes

	rocket.chat-0 stop rendering 1123
	rocket.chat-0 startStreamBroadcast
	rocket.chat-0 {"line":"71","file":"percolate_synced-cron.js","message":"SyncedCron: Scheduled \"Generate and save statistics\" next run @Tue Jan 12 2016 19:18:56 GMT+0000 (UTC)","time":{"$date":1452626336739},"level":"info"}
	rocket.chat-0 Updating login service Accounts_OAuth_Facebook
	rocket.chat-0 Enabling LDAP
	rocket.chat-0 Updating login service Accounts_OAuth_Facebook_id
	rocket.chat-0 Updating login service Accounts_OAuth_Facebook_secret
	rocket.chat-0 Updating login service SAML_Custom_default
	rocket.chat-0 Updating login service Accounts_OAuth_GitHub_Enterprise
	rocket.chat-0 Updating login service Accounts_OAuth_GitHub_Enterprise_id
	rocket.chat-0 Updating login service Accounts_OAuth_GitHub_Enterprise_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Github
	rocket.chat-0 Updating login service Accounts_OAuth_Github_id
	rocket.chat-0 Updating login service Accounts_OAuth_Github_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Gitlab
	rocket.chat-0 Updating login service Accounts_OAuth_Gitlab_id
	rocket.chat-0 Updating login service Accounts_OAuth_Gitlab_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Google
	rocket.chat-0 Updating login service Accounts_OAuth_Google_id
	rocket.chat-0 Updating login service Accounts_OAuth_Google_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Linkedin
	rocket.chat-0 Updating login service Accounts_OAuth_Linkedin_id
	rocket.chat-0 Updating login service Accounts_OAuth_Linkedin_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Meteor
	rocket.chat-0 Updating login service Accounts_OAuth_Meteor_id
	rocket.chat-0 Updating login service Accounts_OAuth_Meteor_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Twitter
	rocket.chat-0 Updating login service Accounts_OAuth_Twitter_id
	rocket.chat-0 Updating login service Accounts_OAuth_Twitter_secret
	rocket.chat-0 Updating login service Accounts_OAuth_Wordpress
	rocket.chat-0 Updating login service Accounts_OAuth_Wordpress_id
	rocket.chat-0 Updating login service Accounts_OAuth_Wordpress_secret
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings
	rocket.chat-0 [publish] settings

**Comment by richardwlu** [comment]
We've restored the mongo database and is currently stable although the original issue with `/tmp/ufs/HHTCTRvN4up7S7tSH` is still prevalent. The avatar uploads are back to GridFS. We actually disabled avatar uploading and this issue still presented itself. The logs for Rocketchat and mongo don't show anything specific as it would be nice to know if it came from a user action.

**Comment by Sing-Li** [comment]
Please re-open if you are still experiencing instability.   Also please be more specific about your configuration - memory size of server running Mongo (and the one running RC), disk size, # CPU, and so on.

**Comment by richardwlu** [comment]
@Sing-Li  We are still having issues. Can you elaborate on "Error: Meteor code must always run within a Fiber"?

Specs:
Mongo is running on the same server as Rocketchat with no replication - would there happen to be a tutorial or guide that helps explain how to get a replication up? I've just started looking into it since we have been having issues.
Linux version 3.10.0-327.el7.x86_64 (mockbuild@x86-034.build.eng.bos.redhat.com) (gcc version 4.8.3 20140911 (Red Hat 4.8.3-9)
No. of CPU: 2
Ram: 8 GB
Disk size:     
    Filesystem               Size  Used Avail Use% Mounted on
    /dev/sda2                 11G  2.7G  7.9G  26% /
    devtmpfs                 3.9G     0  3.9G   0% /dev
    tmpfs                    3.9G     0  3.9G   0% /dev/shm
    tmpfs                    3.9G   17M  3.9G   1% /run
    tmpfs                    3.9G     0  3.9G   0% /sys/fs/cgroup
    /dev/sda3                 11G  1.8G  8.8G  17% /usr
    /dev/sda6                8.3G  787M  7.6G  10% /home
    /dev/mapper/vg_chat-opt   90G   33M   90G   1% /opt
    /dev/sda1                246M  139M   95M  60% /boot
    /dev/sda5                 11G  5.8G  4.8G  55% /var
    tmpfs                    783M     0  783M   0% /run/user/0
    tmpfs                    783M     0  783M   0% /run/user/970100803

**Comment by Sing-Li** [comment]
RE: Error: Meteor code must always run within a Fiber"?
That's exactly what it means, and will not be helpful in diagnosis.  
Situation like this is common in asynchronous system, the cause (or traces of the cause) of the problem may be long gone from the system.

Do you have bursty upload (of large files) happening within your user base?  

Mongo replica-set configuration:
https://docs.mongodb.org/manual/tutorial/deploy-replica-set/

**Comment by richardwlu** [comment]
@Sing-Li I see. Yeah it has been hard to pin point.

The largest file that has been uploaded that I see after running `db.rocketchat_uploads.files.find({length:{$gt: 100000}}, {'filename': 1, 'length': 1, 'chunksize':1, 'uploadDate': 1}).sort ({'length': -1}).sort({length:1});` shows a 2.076 MB file.

I shut off avatar uploading now though since we did have a 4 MB avatar and this was when the default setting was to not resize avatars (wish I caught this earlier). I ran into issues when I switched the avatars to be saved to the FileSystem and to be resized, thus, deciding to shut off avatar uploading completely to see if any issues arise. So far so good for two days.

What would you consider as a bursty/ large file though? The current File Upload max is 2097152 bytes (default).

**Comment by richardwlu** [comment]
To further investigate, I recently enabled Avatar uploads and the system went down a few hours after possibly due to a user uploading a large glorious gif. I restarted rocketchat and switched the Resize Avatars from off to on and the server immediately crashed. We do have GraphicsMagick 1.3.20 installed.

Can anyone shine some light on this situation?

We are currently running on the older v0.8.0 since we are not confident in updating just yet. If this is an issue that has been fixed then we would love to update.
